lib/grpc_reflection/service/builder.ex

```diff
@@ -198,4 +279,102 @@ defmodule GrpcReflection.Service.Builder do
   end
 
   defp upcase_first(<<first::utf8, rest::binary>>), do: String.upcase(<<first::utf8>>) <> rest
+
+  # generate a field descriptor from a field props struct
+  defp convert_to_field_descriptor(
+         extendee,
+         %Protobuf.Extension.Props.Extension{field_props: field_props}
+       ) do
+    {type, type_name} = type_from_field_props(field_props)
+
+    %Google.Protobuf.FieldDescriptorProto{
+      name: field_props.name,
+      number: field_props.fnum,
+      label: label_from_field_props(field_props),
+      type: type,
+      type_name: type_name,
+      extendee: extendee
+    }
+  end
+
+  # Google.Protobuf.FieldDescriptorProto.Label
+  defp label_from_field_props(%Protobuf.FieldProps{optional?: true}), do: 1
+  defp label_from_field_props(%Protobuf.FieldProps{repeated?: true}), do: 3
+  defp label_from_field_props(%Protobuf.FieldProps{required?: true}), do: 2
+
+  # Google.Protobuf.FieldDescriptorProto.Type
+  defp type_from_field_props(%Protobuf.FieldProps{type: type}) do
+    case type do
```

Comment:
These are more proto2 specific rather than builder spec, can you move these utils into a builder submodule to help keep the scope of this module focused?

Reply:
Sure, I moved them to submodule `Util`.

Reply:
You moved them into`GrpcReflection.Service.Util`, but that is not a submodule of `GrpcReflection.Service.Builder`, its a submodule of `GrpcReflection.Service`.  The functions are only used by the Builder module, so it can be nested deeper as `GrpcReflection.Service.Builder.Util` to match the scope and usage.